xray/src/bin/inpaint_xray_quadtree.rs

```diff
@@ -0,0 +1,158 @@
+use fnv::FnvHashSet;
+use quadtree::{Direction, NodeId, SpatialNodeId};
+use std::error::Error;
+use std::fs;
+use std::io;
+use std::path::{Path, PathBuf};
+use structopt::StructOpt;
+use xray::{
+    generation::{assign_background_color, create_non_leaf_nodes, TileBackgroundColorArgument},
+    inpaint::perform_inpainting,
+    utils::{get_image_path, get_meta_pb_path},
+    Meta,
+};
+
+#[derive(StructOpt, Debug)]
+#[structopt(name = "inpaint_xray_quadtrees")]
+/// Inpaint a (possibly partial) xray quadtree. We assume that the input
+/// quadtree was generated with transparent background color, so we can
+/// determine which pixels to actually fill.
+struct CommandlineArguments {
+    /// Directory with the (possibly partial) quadtree to be inpainted.
+    /// Needs to include all leaf nodes of the neighboring quadtrees as well
+    /// for smooth inpainting results.
+    #[structopt(parse(from_os_str))]
+    input_directory: PathBuf,
+    /// Directory where to write the inpainted quadtree. Does *not*
+    /// have to be disjoint from input_directory.
+    #[structopt(parse(from_os_str), long)]
+    output_directory: PathBuf,
+    /// Tile background color.
+    #[structopt(default_value = "white", long)]
+    tile_background_color: TileBackgroundColorArgument,
+    /// The inpainting distance in pixels to fill holes (in particular useful for high resolutions)
+    #[structopt(long)]
+    inpaint_distance_px: u8,
+    /// The root node id to start inpainting with.
+    #[structopt(default_value = "r", long)]
+    root_node_id: NodeId,
+}
+
+fn get_adjacent_leaf_node_ids(
```

Comment:
We could either add a method `neighbor(&self, direction: Direction)` to `NodeId` or create a `neighbor(node_id: NodeId, direction: Direction)` function/closure here and use it twice downstairs.

Reply:
Done